packages/client/src/rpc/validation.ts

```diff
@@ -1,18 +1,21 @@
 import { INVALID_PARAMS } from './error-code.js'
 
+type RPCMethod = (params: any[]) => Promise<any>
```

Comment:
Looks like this is leading to type issues, see this CI run: https://github.com/ethereumjs/ethereumjs-monorepo/actions/runs/12370881134/job/34609917182?pr=3814 
Could you address those, either by improving the type (either this declared type or in usage) or alternatively, remove it?

Reply:
Thanks, I will fix it. To avoid it happening again, is there any precheck I should do before I submit the PR?

Reply:
@gabrocheleau Fixed the type to return any. 
I also noticed several functions don't need to be wrapped with middleware because the middleware function is for parameter validation, and those functions have no parameters. So, I just removed it. Let me know if that works.